### PR TITLE
Update runtime.json with manjaro information

### PR DESCRIPTION
**Pull request by TobiasLaving**
Without specifying runtime on manjaro we cannot run the projects.
Shameless copy of arch entries

I'd like to point out that while /etc/os-release points to "manjaro" and not "manjaro-x64" i have to use the linux-x64 runtime. 
Perhaps manjaro simply does not have non x64 values? ( I have not compared to a arch system, do they have arch/arch-x64?)

FYI: Manjaro has a "ID_LIKE" param in os_release that has arch as input. Perhaps that would be another way forward.

### CODE REVIEW

**Comment by ericstj** [comment]
The mechanism for adding RIDs is described here, can you please follow this?
https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/readme.md#adding-new-rids

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=63338) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=63338) before we can accept your contribution.<br/><br/>:x: TobiasLaving [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=63338)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=63338) it.</sub>

**Comment by TobiasLaving** [comment]
Ah, you are correct

I simply skimmed it and missed all the important stuff.. :) 
I force pushed the commit with proper content

Trying to run it locally to verify the change, need to read up on the build environment though

**Comment by eerhardt** [comment]
Can you post what `$cat /etc/os-release` produces on a manjaro linux machine?

**Comment by TobiasLaving** [comment]
Here you go:
```
[tobias@tobias-expertbook notes]$ cat /etc/os-release 
NAME="Manjaro Linux"
ID=manjaro
ID_LIKE=arch
BUILD_ID=rolling
PRETTY_NAME="Manjaro Linux"
ANSI_COLOR="32;1;24;144;200"
HOME_URL="https://manjaro.org/"
DOCUMENTATION_URL="https://wiki.manjaro.org/"
SUPPORT_URL="https://manjaro.org/"
BUG_REPORT_URL="https://bugs.manjaro.org/"
LOGO=manjarolinux
```

**Comment by vitek-karas** [comment]
> Yet running the application fails with "not supported on this platform".

Could you please paste the exact error you get?
Also (assuming the error is coming from the host), could you run the failing app with `export COREHOST_TRACE=1` and `export COREHOST_TRACEFILE=host.txt` - and then share the `host.txt` (please be aware that the file contains local machine environment and paths and other info).

**Comment by TobiasLaving** [comment]
Thanks for the quick help guys!
Sorry for my slow responses

The runtime was installed using manjaro package manager pamac
```
[tobias@tobias-expertbook Api]$ pamac list | grep runtime
aspnet-runtime                    6.0.0.sdk100-2              community  20,2 MB
dotnet-runtime                    6.0.0.sdk100-2              community  68,0 MB
```

[here](https://github.com/dotnet/runtime/files/7848666/temp.txt) is the full printout of the error, seems to originate at the sql client libary

Here is the tracefile:  [here](https://github.com/dotnet/runtime/files/7848710/host.txt)
I also include a tracefile when specifying the runtime ( Which works ): [host2.txt](https://github.com/dotnet/runtime/files/7848725/host2.txt)
 
 To my untrained eye, it looks like the first run does properly resolve the runtime..?

**Comment by vitek-karas** [comment]
The error is coming from `System.Data.SqlClient.dll` - it basically means that the system was not able to pick the right platform specific implementation of this dll, and it fell back to the default - which simply always throws.

The underlying issue is this:

The RID is detected as:
```
HostRID is manjaro-x64
Falling back to base HostRID: linux-x64
```

So for some reason it doesn't recognize `manjaro-x64` as valid RID - and thus it falls back to the generic `linux-x64`.
But at the same time the RID fallback graph is just this:
```
The rid fallback graph is: {
arch-x64 => [
arch, 
linux-x64, 
linux, 
unix-x64, 
unix, 
any, 
base, 
]
}
```

This is not enough - it only tells the host what to do if it detect the RID as `arch-x64`, it doesn't have the rest. For example there should be `linux-x64 => [ ... ] ` section and so on.
Due to this, when it tries to resolve the right asset for `System.Data.SqlClient` it fails to do so and generates this (hidden) warning:
```
The targeted framework does not support the runtime 'linux-x64'. Some native libraries from [Microsoft.Data.SqlClient/2.0.1] may fail to load on this platform.
```

Please look into `/usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.0/Microsoft.NETCore.App.deps.json` and see what's in the RID graph there...

The reason why it works in the second case - that is you add `--runtime arch-x64` on the command line - this makes `dotnet build` (which is basically a part of `dotnet run`) to produce an OS specific build of the app. So the resolution of which particular version of a dll to pick is done by NuGet during the build - and since it has the full graph (always, part of SDK/NuGet) it correctly resolves the `unix` RID specific asset for `System.Data.Client.dll`. The host has no work in this case as there are no RID specific assets in the final app. You will see this - in the first case (failure) there should be `runtime` subdirectory in the built output, in the second case (success) there should be no subdirectory.

**Comment by TobiasLaving** [comment]
Thanks @vitek-karas for that explanation

It makes sense that it fails to use manjaro-x64 as RID, the PR actually try to add it. ( All of the above commands was done with the offical application )

Why is HostRID resolved to manjaro-x64 though, if `dotnet --info` show RID listed as arch-x64? I kind of assumed the same code/method of detection would be used.

My [Microsoft.NETCore.App.deps.json.txt](https://github.com/dotnet/runtime/files/7849046/Microsoft.NETCore.App.deps.json.txt)
 is indeed missing a linux-x64 entry, why is that?

**Comment by vitek-karas** [comment]
This is really weird - I tried locally and `-r arch64` will produce the super short RID graph you see. But using `-r linux-x64` produces a long RID graph with a lot of linux distros in it.
I honestly don't know how this works - that's more an SDK question than runtime/host. We would also need to check what is the RID graph we ship in the various versions of the runtime packs for `Microsoft.NETCore.App`.

Similarly for the RID displayed by `dotnet --info` -- that's an SDK thing as well. It should agree with host, but apparently it doesn't.

Please note that this seems to be broken regardless of your changes. Since you have all the samples and such, could you please create an SDK issue? I'll be happy to add whatever details I found on my end.

We will need somebody to dig into the SDK side of things here.

There's a way to override the RID detection in host - set the env. variable `RUNTIME_ID` to the RID you want - it will still check it against the RID graph though. So in your can I think it should work if you set `RUNTIME_ID=arch-x64`...

**Comment by TobiasLaving** [comment]
Ok, will do!
Can you link we to the right place?

Btw, RUNTIME_ID does not seem to work, atleast not when using `dotnet run`or rider as IDE
Do you know of any other way to configure, rather than setting RuntimeIdentifier in csproj?

(Also, should I close the PR or what is the next step?)

**Comment by vitek-karas** [comment]
The SDK repo is dotnet/sdk.

Can you try running the app by `dotnet build` and the just run the executable created in the output - with `RUNTIME ID`?

I think the PR looks good - you should try to validate that it helps - meaning building with these changes, producing the shared framework and trying that on the manjaro machine if it can pick up the correct RID. Validating integration with SDK (like being able to build for manjaro RID itself).

**Comment by eerhardt** [comment]
> Similarly for the RID displayed by dotnet --info -- that's an SDK thing as well. It should agree with host, but apparently it doesn't.

That code is [here](https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/dotnet/CommandLineInfo.cs#L35-L46). Is the version that manjaro distributes built from an `arch` machine? That would explain this behavior.

**Comment by TobiasLaving** [comment]
On that topic, I am having some difficulties testing the changes
I've managed to build the dotnet binary using a docker container, but it report net7 as framework and won´t run the dll
Is there any guide how to build and run a new dotnet binary? I cannot find a concrete guide, but perhaps I simply missed it

@eerhardt That could very well be the reason, the only thing that makes sense

**Comment by vitek-karas** [comment]
To run an app with your locally built runtime, this is what I do (there might be better ways):
* build everything `build.sh`/`build.cmd` - there is probably some subset which would be sufficient, but this is just easier
* there should be a directory similar to this one `artifacts\bin\testhost\net7.0-windows-Debug-x64` (adapt to your platform/arch). This is a proper runtime layout which can run apps - so you should find `dotnet.exe` there and so on. (no SDK)
* Build a sample app with your usual SDK (.NET 6 is enough) - in the app's project file add `<RollForward>Major</RollForward>` (this will allow the app to run on .NET 7)
* Run the app either via `artifacts\bin\testhost\net7.0-windows-Debug-x64\dotnet app.dll` or by setting `DOTNET_ROOT=artifacts\bin\testhost\net7.0-windows-Debug-x64` and then running the executable.

This doesn't validate the SDK integration - for example `dotnet --info` works somewhat differently if there's no SDK, but it should validate the host/runtime changes. So you can test things like what is RID recognized by the host (COREHOST_TRACE for that), and what is the RID reported by the framework, what is the OS name reported by framework and so on.

**Comment by TobiasLaving** [comment]
Thanks a ton @vitek-karas !

I have now tested and verified that the above commit enables me to run with manjaro straight out of the box.
Would be great if we could get it in!

Here is a link to the issue in sdk: https://github.com/dotnet/sdk/issues/23390

File: src/libraries/Microsoft.NETCore.Platforms/src/runtime.json
Comment (TobiasLaving) [reply src/libraries/Microsoft.NETCore.Platforms/src/runtime.json:1732]: Yes, I did edit manually initally

I simply assumed the runtime.json would be in the correct format after `dotnet pack` without any need to change anything. I cleaned it up and updated. Good catch.

**Review by omajid — APPROVED** [review]
Looks good to me now. Thanks!

**Comment by TobiasLaving** [comment]
@vitek-karas Looks like I need a review from you as well :)

**Review by vitek-karas — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]

**Comment by vitek-karas** [comment]
@TobiasLaving please sign the CLA per this comment https://github.com/dotnet/runtime/pull/63338#issuecomment-1007410513

@eerhardt - the failures in CI seem unrelated (timeouts on OSX), I think we're OK to merge after the CLA is done. What do you think?

**Comment by TobiasLaving** [comment]
@vitek-karas Ah, I missed that
Done!

**Pull request merged**
